netman/adapters/switches/juniper/base.py

```diff
@@ -113,6 +113,32 @@ def get_vlans(self):
 
         return vlan_list
 
+    def get_vlan(self, number):
+        config = self.query(one_vlan_by_vlan_id(number), all_interfaces)
+        if len(config.xpath("data/configuration/vlans/vlan")) >= 1:
+            vlan_node = config.xpath("data/configuration/vlans/vlan")[0]
+            number_node = first(vlan_node.xpath("vlan-id"))
```

Comment:
You could probably have methos doing this and have get_vlans use it too... this seems like a lot of duplicate code